src/arbcall.jl

```diff
@@ -58,72 +55,85 @@ end
 function Carg(str)
     m = match(r"(?<const>const)?\s*(?<type>\w+(\s\*)?)\s+(?<name>\w+)", str)
     isnothing(m) && throw(ArgumentError("string doesn't match c-argument pattern"))
-
-    return Carg{arbargtypes[m[:type]]}(m[:name], !isnothing(m[:const]))
+    cnst =
+        !isnothing(m[:const]) ||
+        (!isnothing(m[:type]) && (m[:type] == "arb_srcptr" || m[:type] == "acb_srcptr"))
+    return Carg{arbargtypes[m[:type]]}(m[:name], cnst)
 end
 
 name(ca::Carg) = ca.name
 isconst(ca::Carg) = ca.isconst
 
-rawtype(::Carg{T}) where T = T
+rawtype(::Carg{T}) where {T} = T
 
 jltype(ca::Carg) = rawtype(ca)
 jltype(ca::Carg{Cint}) = Integer
 jltype(ca::Carg{Clong}) = Integer
 jltype(ca::Carg{Culong}) = Unsigned
 jltype(ca::Carg{Cdouble}) = Base.GMP.CdoubleMax
-jltype(ca::Carg{arb_rnd}) = Union{arb_rnd, RoundingMode}
-jltype(ca::Carg{Base.MPFR.MPFRRoundingMode}) = Union{Base.MPFR.MPFRRoundingMode, RoundingMode}
+jltype(ca::Carg{arb_rnd}) = Union{arb_rnd,RoundingMode}
+jltype(ca::Carg{Base.MPFR.MPFRRoundingMode}) =
+    Union{Base.MPFR.MPFRRoundingMode,RoundingMode}
 jltype(ca::Carg{Cstring}) = AbstractString
 jltype(ca::Carg{Vector{Clong}}) = Vector{<:Integer}
 jltype(ca::Carg{Vector{Culong}}) = Vector{<:Unsigned}
-jltype(::Carg{T}) where {T <: Union{Mag, Arf, Arb, Acb}}  = Union{T, cstructtype(T), Ptr{cstructtype(T)}}
+jltype(ca::Carg{ArbVector}) = Union{ArbVector,cstructtype(ArbVector)}
+jltype(ca::Carg{AcbVector}) = Union{AcbVector,cstructtype(AcbVector)}
```

Comment:
This was in Marek's draft 
```julia
 Union{AcbVector,cstructtype(AcbVector),Ptr{acb_struct}}
```
But the `Ptr{acb_struct}` part could lead to ambiguities in dispatch. Also the default  of `len` didn't work.